// morrow-common/dist/index.js
"use strict";
var __importDefault = (this && this.__importDefault) || function (mod) {
    return (mod && mod.__esModule) ? mod : { "default": mod };
};
Object.defineProperty(exports, "__esModule", { value: true });
exports.environmentVariables = exports.authenticate = exports.ProjectRequest = exports.ProjectsResponse = exports.ProjectServiceService = exports.ProjectServiceClient = exports.TeamResponse = exports.UserServiceClient = exports.UserRequest = exports.UserServiceService = exports.JWTService = exports.delay = void 0;
var delay_1 = require("./library/delay");
Object.defineProperty(exports, "delay", { enumerable: true, get: function () { return delay_1.delay; } });
var jwt_1 = require("./jwt/jwt");
Object.defineProperty(exports, "JWTService", { enumerable: true, get: function () { return jwt_1.JWTService; } });
var cmn_1 = require("./grpc/cmn");
Object.defineProperty(exports, "UserServiceService", { enumerable: true, get: function () { return cmn_1.UserServiceService; } });
Object.defineProperty(exports, "UserRequest", { enumerable: true, get: function () { return cmn_1.UserRequest; } });
Object.defineProperty(exports, "UserServiceClient", { enumerable: true, get: function () { return cmn_1.UserServiceClient; } });
Object.defineProperty(exports, "TeamResponse", { enumerable: true, get: function () { return cmn_1.TeamResponse; } });
Object.defineProperty(exports, "ProjectServiceClient", { enumerable: true, get: function () { return cmn_1.ProjectServiceClient; } });
Object.defineProperty(exports, "ProjectServiceService", { enumerable: true, get: function () { return cmn_1.ProjectServiceService; } });
Object.defineProperty(exports, "ProjectsResponse", { enumerable: true, get: function () { return cmn_1.ProjectsResponse; } });
Object.defineProperty(exports, "ProjectRequest", { enumerable: true, get: function () { return cmn_1.ProjectRequest; } });
const dotenv_1 = __importDefault(require("dotenv"));
var authMiddleware_1 = require("./middlewares/authMiddleware");
Object.defineProperty(exports, "authenticate", { enumerable: true, get: function () { return authMiddleware_1.authenticate; } });
dotenv_1.default.config();
exports.environmentVariables = {
    CLOUDINARY_URL: process.env.CLOUDINARY_URL,
    CLOUDINARY_NAME: process.env.CLOUDINARY_NAME,
    CLOUDINARY_APIKEY: process.env.CLOUDINARY_APIKEY,
    CLOUDINARY_APISECRET: process.env.CLOUDINARY_APISECRET,
};
